packages/http/src/mocker/HttpMocker.ts

```diff
@@ -1,82 +1,106 @@
-import { IMocker, IMockerOpts } from '@stoplight/prism-core';
-import { Dictionary, IHttpHeaderParam, IHttpOperation, INodeExample, INodeExternalExample } from '@stoplight/types';
+import { IMocker, IMockerOpts, IPrismInput } from '@stoplight/prism-core';
+import { Dictionary, IHttpHeaderParam, IHttpOperation, INodeExample } from '@stoplight/types';
 
 import * as caseless from 'caseless';
+import { Either } from 'fp-ts/lib/Either';
+import { Reader } from 'fp-ts/lib/Reader';
 import { isEmpty, isObject, keyBy, mapValues } from 'lodash';
+import { Logger } from 'pino';
 import {
+  ContentExample,
   IHttpConfig,
   IHttpOperationConfig,
   IHttpRequest,
   IHttpResponse,
   PayloadGenerator,
   ProblemJsonError,
 } from '../types';
+import withLogger from '../withLogger';
 import { UNPROCESSABLE_ENTITY } from './errors';
 import { generate, generateStatic } from './generator/JSONSchema';
 import helpers from './negotiator/NegotiatorHelpers';
 import { IHttpNegotiationResult } from './negotiator/types';
 
-export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpConfig, IHttpResponse> {
+export class HttpMocker
+  implements IMocker<IHttpOperation, IHttpRequest, IHttpConfig, Reader<Logger, Either<Error, IHttpResponse>>> {
   public mock({
     resource,
     input,
     config,
-  }: Partial<IMockerOpts<IHttpOperation, IHttpRequest, IHttpConfig>>): IHttpResponse {
-    let payloadGenerator: PayloadGenerator = generateStatic;
+  }: IMockerOpts<IHttpOperation, IHttpRequest, IHttpConfig>): Reader<Logger, Either<Error, IHttpResponse>> {
+    const payloadGenerator: PayloadGenerator =
+      config && typeof config.mock !== 'boolean' && config.mock.dynamic ? generate : generateStatic;
 
-    if (config && typeof config.mock !== 'boolean' && config.mock.dynamic) {
-      payloadGenerator = generate;
-    }
+    return withLogger(logger => {
+      // setting default values
```

Comment:
would it be much work to make this a separate function? Just like `negotiateResponse` ?